### PR TITLE
DEVOPS-349 - GPR -> GHCR Migration

### PR DESCRIPTION
**Pull request by jsfillman**
Required migration from GitHub's Package Registry to the newer Container Service.

Primary Advantages are:

1. Allows unauthenticated pulls of public images (needed for our workflow)
2. Simplifies package naming & management
 3. Improved org-wide package UI

Once merged, you can see any future dev and prod images in the GitHub UI here:

- https://github.com/orgs/kbase/packages/container/package/narrative-develop
- https://github.com/orgs/kbase/packages/container/package/narrative

### New Image URLs

| Evironment  | Image URL   |
| ----------- | ----------- |
| CI          | ghcr.io/kbase/narrative-develop:latest |
| Appdev      | ghcr.io/kbase/narrative:appdev         |
| Next        | ghcr.io/kbase/narrative:next           |
| Production  | ghcr.io/kbase/narrative:prod           |

### CODE REVIEW

**Comment by briehl** [comment]
Codacy's whiny about not putting quotes around your `echo $DOCKER_TOKEN ...` line. But I think that's safe to ignore.

File: .github/workflows/build_test_pr.yaml [hunk 1]
```diff
@@ -22,6 +22,6 @@ jobs:
       env:
         PR: "${{ github.event.pull_request.number }}"
         SHA: "${{ github.event.pull_request.head.sha }}"
-        DOCKER_ACTOR: "${{ github.actor }}"
-        DOCKER_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+        DOCKER_ACTOR: "jsfillman"
```
Comment (briehl) [inline .github/workflows/build_test_pr.yaml:25]: So if I understand this right, all the Docker actions will be done as your username? And not a configured actor?

File: docs/deploy_narrative.md [hunk 2]
```diff
@@ -37,6 +37,8 @@ Simply restart the `narrative` container for CI in Rancher. No other action is n
 Once a pull request is merged from `develop` to `master` a new _production_ image will be available at:
 `docker.pkg.github.com/kbase/narrative/narrative:latest`.
```
Comment (briehl) [inline docs/deploy_narrative.md:38]: Will this line need to update from `docker.pkg.github.com` -> `ghcr.io`, too?

File: docs/deploy_narrative.md [hunk 3]
```diff
@@ -51,14 +53,14 @@ Once a pull request is merged from `develop` to `master` a new _production_ imag
     -t IMAGE_TAG is the *current* Docker image tag, typically `pr-#` or `latest`
     -e TARGET Sets target environment's tag. This is typically either:`appdev`, `next`, or `prod`.
 ```
-4. Restart the `narrative` container for the desired environment (e.g. `appdev`) in Rancher.
+4. Upgrade the `narrative` container for the desired environment (e.g. `next` ) in Rancher.
 
 
 ### Image URLs
 
 | Evironment  | Image URL   |
```
Comment (briehl) [inline docs/deploy_narrative.md:61]: Evironment -> Environment (I know it's not part of your changes, but I just stumbled on it...

**Review by briehl — COMMENTED** [review]
I think it's fine, you're just rerouting the package store to ghcr.io. Got a couple little questions/comments, though.

File: .github/workflows/build_test_pr.yaml (hunk 1 above)
Comment (jsfillman) [reply .github/workflows/build_test_pr.yaml:25]: Unfortunately, GitHub Container Registry can't "inherit" temporary auth tokens the way GitHub Package Registry can (yet).

This is a temporary solution until we can get a service account going to run CI independently of any specific user. Let's try to push for service account creation during Monday's standup, but it'll be a very easy find and replace to update this.

**Comment by jsfillman** [comment]
Due to some of the oddities of Actions & quoting, the Codacy recommendation actually breaks the build script.
Reverting and ensured builds are passing, and all links are updated as discussed.
@briehl - Feel free to take one last look, but this should be good to go!

**Review by briehl — APPROVED** [review]
Looks good. I just want to make a note saying that we'll eventually update the user (`jsfillman`) to a generic devops user that has access.

Until then, @jsfillman needs to not play in traffic. 😉

**Pull request merged**